core/server/main.cpp

```diff
@@ -147,6 +151,42 @@ int main(int a_argc, char **a_argv) {
 
         return 0;
       }
+      if (cfg_log_level != UINT_MAX) {
+        if (cfg_log_level >=
+            static_cast<unsigned int>(LogLevel::LAST_SENTINEL)) {
+          DL_ERROR(log_context,
+                   "Invalid log level provided, defaulting to INFO");
+        } else {
+          LogLevel cast_log_level = static_cast<LogLevel>(cfg_log_level);
+          std::string str_log_level = "";
+          switch (cast_log_level) {
+          case LogLevel::CRITICAL:
+            str_log_level = "CRITICAL";
+            break;
+          case LogLevel::ERROR:
+            str_log_level = "ERROR";
+            break;
+          case LogLevel::WARNING:
+            str_log_level = "WARNING";
+            break;
+          case LogLevel::INFO:
+            str_log_level = "INFO";
+            break;
+          case LogLevel::DEBUG:
+            str_log_level = "DEBUG";
+            break;
+          case LogLevel::TRACE:
+            str_log_level = "TRACE";
+            break;
+          case LogLevel::LAST_SENTINEL:
+            str_log_level = "INVALID"; // should never be reached
+            break;
+          }
+          std::string log_message = "Setting log level to " + str_log_level;
+          DL_INFO(log_context, log_message);
+          global_logger.setLevel(cast_log_level);
```

Comment:
Actually won't this be confusing if str_log_level is INVALID but we are defaulting to INFO?

Reply:
Is the last case just there in case someone changes th code, will it be hit ever since you have an else checking that the SENTINEL value is not used?

Reply:
Yes! I changed it to throw an exception in case we hit an invalid state

Reply:
I added the last case so that every case is explicitly defined, and if the `LogLevel` enum is ever changed, a compiler error will be thrown to update the switch as well. I'm not sure this is my favorite approach, but it will give some extra safety over `default`